newrelic-agent/src/main/java/com/newrelic/agent/config/SystemPropertyProvider.java

```diff
@@ -128,21 +110,6 @@ private void addPropertyWithoutSystemPropRoot(Map<String, Object> nrProps, Strin
     }
 
     public String getEnvironmentVariable(String key) {
-        String propVal = envVars.get(key);
-        if (propVal != null) {
-            return propVal;
-        }
-
-        return getenv(key);
-    }
-
-    private String getenv(String key) {
-        String val = environmentFacade.getenv(key);
-        if (val != null) {
-            return val;
-        }
```

Comment:
This is the conditional that was retrieving the dotted name from the environment. The simplest change would be to remove this conditional alone.